common/src/main/java/com/microsoft/identity/common/adal/internal/cache/StorageHelper.java

```diff
@@ -497,147 +526,108 @@ protected void setBlobVersion(@NonNull String blobVersion) {
 
             case KEYSTORE_ENCRYPTED_KEY:
-                Logger.verbose(TAG + methodName, "Loading KeyStore encrypted key.");
-                return loadKey();
+                return loadKeyStoreEncryptedKey();
         }
 
         Logger.verbose(TAG + methodName, "Unknown KeyType. This code should never be reached.");
         throw new GeneralSecurityException(ErrorStrings.UNKNOWN_ERROR);
     }
 
     /**
-     * If needed, get the key from inactive broker (if there is one).
-     * If it fails to get a new key. It will create a new one.
-     *
-     * IMPORTANT: This function can only be invoked by Broker apps, and must NOT be called from main thread.
+     * Encrypt the given unencrypted symmetric key with Keystore key and save to storage.
      */
-    public void migrateEncryptionKeyIfNeeded() throws GeneralSecurityException, IOException {
-        final String methodName = ":migrateEncryptionKeyIfNeeded";
-
-        if (!(COMPANY_PORTAL_APP_PACKAGE_NAME.equalsIgnoreCase(getPackageName()) ||
-                AZURE_AUTHENTICATOR_APP_PACKAGE_NAME.equalsIgnoreCase(getPackageName()))) {
-            final String errorMessage = "Caller is not a broker. Migration is not needed";
-            Logger.error(TAG + methodName, errorMessage, null);
-            throw new IllegalStateException(errorMessage);
-        }
-
-        mEncryptionKey = loadKey();
-        if (mEncryptionKey != null){
-            Logger.verbose(TAG + methodName, "Key already exists.");
-            return;
-        }
-
-        mEncryptionKey = getKeyFromInactiveBroker();
-        if (mEncryptionKey != null){
-            Logger.verbose(TAG + methodName, "Key is successfully retrieved from inactive broker. Saving key...");
-            saveSecretKey(mEncryptionKey);
-            return;
-        }
-
-        Logger.verbose(TAG + methodName, "Key migration failed. Create a new key.");
-        mEncryptionKey = createKey();
-        return;
-    }
-
-    @Nullable
-    protected SecretKey getKeyFromInactiveBroker() {
-        final String methodName = ":getKeyFromInactiveBroker";
-        final String serializedKey = InactiveBrokerClient.getSerializedSymmetricKeyFromInactiveBroker(mContext, getPackageName());
-
-        if (serializedKey == null || serializedKey.length() == 0) {
-            Logger.verbose(TAG + methodName, "The returned bundle doesn't contain any key.");
-            return null;
-        }
-
-        return deserializeSecretKey(serializedKey);
-    }
-
-    private void saveSecretKey(@NonNull SecretKey key) throws GeneralSecurityException, IOException {
+    public void saveKeyStoreEncryptedKey(@NonNull SecretKey unencryptedKey) throws GeneralSecurityException, IOException {
         final String methodName = ":saveSecretKey";
 
-        Logger.verbose(TAG + methodName, "Saving secret key to storage.");
+        try {
+            logFlowStart(methodName, AuthenticationConstants.TelemetryEvents.KEYCHAIN_WRITE);
```

Comment:
Why is this a `KEYCHAIN_WRITE` event here? , shouldn't this be when create a asymetric key pair in `generateKeyPairFromAndroidKeyStore`  , this incorrectly could mark any exceptions here in methods `wrap` and `writeKeyData` as key store related failures right ?